Dockerfile.develop

```diff
@@ -0,0 +1,45 @@
+FROM node:10-alpine
+
+# Support custom branches of the react-sdk and js-sdk. This also helps us build
+# images of riot-web develop.
+ARG USE_CUSTOM_SDKS=false
+ARG REACT_SDK_REPO="https://github.com/matrix-org/matrix-react-sdk.git"
```

Comment:
we should actually be using locally-linked react-sdk and such when doing development, as the bulk of Riot is in the react-sdk.

Reply:
Working on it :+1:

Reply:
Yes, it is, otherwise the container would *require* people to mount files to certain location. Currently it by default pulls the repos, however allows to override this behavior. This way not only you can choose what repo/folder to use, but also allows you to have only this project as editable without even pulling the SDKs. See [this change](https://github.com/vector-im/riot-web/pull/10794/commits/c95e7a866d4ca05fcc63b90371a1604c9b002be2#diff-04c6e90faac2675aa89e2176d2eec7d8R355).

Reply:
for a development environment where 95% of the app is in the react-sdk, I would definitely expect that people have a local copy of the react-sdk.

Reply:
The current approach works for both cases and at the same time pulls the app dependencies first which means that there already are dependencies for react-sdk and riot-web already cached in the image itself. That means a user that mounts a locally cloned react-sdk does not have to wait long for the installation because the only thing that's going to install with `docker run` is riot-web + different versions of deps that are not cached. This is good for switching between branches since docker container by default does not save the changes in its filesystem(image) unless `docker commit` is used.

Reply:
The installation should be pretty quick if the dependencies have already been pulled the first time. Unfortunately the environment doesn't provide much value if most of the app isn't editable.

Reply:
Docker does not work that way, it does not preserve anything that happens in the container which means every `docker run` command requires the installation from scratch i.e. pulling each dependency. No quick install because the pull would happen every time. That's why I left the original repositories there so the docker *image* already contains the cached dependencies. Only image preserves the state, not container. If I remove the repositories from Dockerfile, everything will be slowed down because the whole fetching/hashing/installing/linking/preparing process would need to be executed in full on each `docker run`. Currently if you do `docker build` the image (which contains the state used by the container) will have all the deps + a default installation according to that ARGS instruction, however you don't need to care about that because as said in the linked instructions from README.md with `docker run` you can specify to use custom, locally available, repositories/folders which will then **replace** the default installation from the image and will be both used for the next installation run with `docker run` command (installs only the new SDKs, uses cached deps, i.e. the case you describe) and will be editable from the host machine i.e. if I change something in react-sdk and press F5 the changes will be immediately reflected in the container. If anything goes wrong a user can then use `docker commit` to create a *new* image from the container and e.g. send it to you/other devs or just share with colleagues.